tomviz/PresetDialog.cxx

```diff
@@ -0,0 +1,39 @@
+/* This source file is part of the Tomviz project, https://tomviz.org/. 
+   It is released under the 3-Clause BSD License, see "LICENSE". */
+
+#include "PresetDialog.h"
+#include "PresetModel.h"
+#include "ui_PresetDialog.h"
+
+#include <QTableView>
+#include <QtWidgets>
+
+namespace tomviz {
+
+PresetDialog::PresetDialog(QWidget* parent)
+  : QDialog(parent), m_ui(new Ui::PresetDialog)
+{
+  m_ui->setupUi(this);
+
+  /*
+    Create instance of PresetModel and pass a pointer to it to the view.
+    The view will then invoke the methods of the model pointer to 
+    determine the number of rows and columns that should be displayed.
+  */
+  QTableView *view = new QTableView(this);
+  PresetModel *model = new PresetModel();
+  view->setModel(model);
+  // make columns fit what is in them
+  view->resizeColumnsToContents();
+  // make columns stetch to fit table size
+  view->horizontalHeader()->setSectionResizeMode(QHeaderView::Stretch);
+
+  // create a grid to center the column
+  QVBoxLayout *layout = new QVBoxLayout;
+  layout->addWidget(view);
+  layout->addWidget(m_ui->buttonBox);
+  this->setLayout(layout);
```

Comment:
While widgets like the QTableView and QVBoxLayout *can* be added in programmatically, we typically would add them in via the Qt Designer GUI. I can help show how if needed.

Reply:
I would rather keep it programmatic for now, Patrick is right that we can use UI files too.